diff --git a/README.md b/README.md
index b4f8986..734f5e9 100644
--- a/README.md
+++ b/README.md
@@ -159,6 +159,7 @@ pip install 'pillow<7'
 cd ~
 git clone https://github.com/NVIDIA-AI-IOT/torch2trt
 cd torch2trt
+git checkout d1fa6f9f20c6c4c57a9486680ab38c45d0d94ec3   # 動作確認済みのバージョン（Wed Nov 4時点）に戻す
 sudo python setup.py install
 sudo python3 setup.py install
 
@@ -188,7 +189,7 @@ cd nano_build_opencv
 | ---- | ---- | ---- | ---- |
 |  pytorch  |  ~~1.6.0~~ 1.4.0  |  1.4.0  |  -  |
 |  torchvision  |  0.2.2  |  0.2.2  |  -  |
-|  torch2trt  |  -  |  -  |  -  |
+|  torch2trt  |  -  |  -  |  動作確認済みのバージョン `git checkout d1fa6f9f20c6c4c57a9486680ab38c45d0d94ec3`  |
 |  sklearn  |  0.23.2  |  Not_Installed  |  -  |
 |  pandas  |  0.22.0 ~~(1.1.3必須かも)~~  |  Not_Installed  |  -  |
 |  cv2  |  3.4.10  |  3.4.10   |  -  |
diff --git a/docker/jetson/start.sh b/docker/jetson/start.sh
index ec0dad7..1841168 100755
--- a/docker/jetson/start.sh
+++ b/docker/jetson/start.sh
@@ -2,6 +2,7 @@
 
 # torch2trt setup
 pushd /tmp/torch2trt
+git checkout d1fa6f9f20c6c4c57a9486680ab38c45d0d94ec3
 python setup.py install
 python3 setup.py install
 popd
diff --git a/scripts/setup/auto_setup.sh b/scripts/setup/auto_setup.sh
index 87cd1d1..72de102 100755
--- a/scripts/setup/auto_setup.sh
+++ b/scripts/setup/auto_setup.sh
@@ -120,6 +120,7 @@ function install_torch2trt(){
     sudo rm -rf torch2trt
     git clone https://github.com/NVIDIA-AI-IOT/torch2trt
     cd torch2trt
+    git checkout d1fa6f9f20c6c4c57a9486680ab38c45d0d94ec3
     sudo python setup.py install
     sudo python3 setup.py install
 }
diff --git a/scripts/setup/auto_setup_for_Linux.sh b/scripts/setup/auto_setup_for_Linux.sh
index 040a079..ec2d9d0 100755
--- a/scripts/setup/auto_setup_for_Linux.sh
+++ b/scripts/setup/auto_setup_for_Linux.sh
@@ -125,6 +125,7 @@ function install_torch2trt(){
     sudo rm -rf torch2trt
     git clone https://github.com/NVIDIA-AI-IOT/torch2trt
     cd torch2trt
+    git checkout d1fa6f9f20c6c4c57a9486680ab38c45d0d94ec3
     sudo python setup.py install
     sudo python3 setup.py install
 }
